OpenRA.Game/Server/Server.cs

```diff
@@ -1163,6 +1163,16 @@ public Session.Client GetClient(Connection conn)
 			return LobbyInfo.ClientWithIndex(conn.PlayerIndex);
 		}
 
+		/// <Remarks> Does not check if client is admin </Remarks>
+		public bool CanKickClient(Session.Client client)
+		{
+			if (State != ServerState.GameStarted || client.IsObserver)
+				return true;
+
+			var player = worldPlayers.FirstOrDefault(p => p?.ClientIndex == client.Index);
+			return player != null && player.Outcome != WinState.Undefined;
```

Comment:
the gui may not offer the option, but it may be good to check if you are no trying to kick yourself

`&& player != Admin|Host && player != self?`

happens often enough the people deadmin themselves.

Reply:
It checks for admin, see the kick code in LobbyCommands.cs

Reply:
Technically it does only check if the admin executes the command, not if the admin is the client to kick. So an admin could kick themselves.

Reply:
Added the check